ocs/agents/registry/agent.py

```diff
@@ -115,7 +115,33 @@ def _register_heartbeat(self, _data):
         addr = feed['agent_address']
         if addr not in self.registered_agents:
             self.registered_agents[addr] = RegisteredAgent(feed)
+        
+        reg_agent = self.registered_agents[addr]
+        publish = op_codes != reg_agent.op_codes
         self.registered_agents[addr].refresh(op_codes=op_codes)
+        if publish:
+            self._publish_agent_ops(reg_agent)
+
+    def _publish_agent_ops(self, reg_agent):
+        addr = reg_agent.agent_address
+        msg = {'block_name': addr,
+               'timestamp': time.time(),
+               'data': {}}
+        self.log.info(addr)
```

Comment:
Maybe more appropriately a debug log? If not maybe some more info should be included, like "heartbeat for {addr} published" or similar?